packages/web3-plugin-example/package.json

```diff
@@ -0,0 +1,63 @@
+{
+	"name": "web3-plugin-example",
+	"version": "0.0.1-alpha.0",
+	"description": "",
+	"main": "dist/index.js",
+	"repository": "https://github.com/ChainSafe/web3.js",
+	"engines": {
+		"node": ">=12.0.0"
+	},
+	"author": "ChainSafe Systems",
+	"license": "LGPL-3.0",
+	"keywords": [
+		"Ethereum",
+		"JavaScript",
+		"API"
+	],
+	"files": [
+		"dist/**/*"
+	],
+	"scripts": {
+		"clean": "rimraf dist",
+		"prebuild": "rimraf dist",
+		"build": "tsc --build",
+		"build:web": "npx webpack",
+		"build:check": "node -e \"require('./dist')\"",
+		"lint": "eslint --ext .js,.ts .",
+		"lint:fix": "eslint --fix --ext .js,.ts .",
+		"format": "prettier --write '**/*'",
+		"test": "jest --config=./test/unit/jest.config.js",
+		"test:coverage": "jest --config=./test/unit/jest.config.js --coverage=true --coverage-reporters=text",
+		"test:ci": "jest --coverage=true --coverage-reporters=json --verbose",
+		"test:watch": "npm test -- --watch",
+		"test:unit": "jest --config=./test/unit/jest.config.js",
+		"test:integration": "jest --config=./test/integration/jest.config.js --runInBand --forceExit",
+		"test:blackbox:ganache:http": "./scripts/black_box_test.sh ganache http",
+		"test:blackbox:ganache:ws": "./scripts/black_box_test.sh ganache ws",
+		"test:blackbox:geth:http": "./scripts/black_box_test.sh geth http",
+		"test:blackbox:geth:ws": "./scripts/black_box_test.sh geth ws",
+		"test:blackbox:infura:http": "./scripts/black_box_test.sh infura http",
+		"test:blackbox:infura:ws": "./scripts/black_box_test.sh infura ws"
+	},
+	"devDependencies": {
+		"@types/jest": "^28.1.6",
+		"@typescript-eslint/eslint-plugin": "^5.30.7",
+		"@typescript-eslint/parser": "^5.30.7",
+		"eslint": "^8.20.0",
+		"eslint-config-prettier": "^8.5.0",
+		"eslint-config-web3-base": "0.1.0",
+		"eslint-plugin-import": "^2.26.0",
+		"jest": "^28.1.3",
+		"jest-extended": "^3.0.1",
+		"prettier": "^2.7.1",
+		"ts-jest": "^28.0.7",
+		"typescript": "^4.7.4"
+	},
+	"dependencies": {
```

Comment:
should those be peerDependencies?

Reply:
I believe `web3` can be a peer dependency, but the other packages are actually being used by the plugin - perhaps this will change a little after I implement @jdevcs's [feedback](https://github.com/ChainSafe/web3.js/pull/5393#discussion_r958579938)

Reply:
I'm not sure they can be since the plugin is actually [importing the packages](https://github.com/ChainSafe/web3.js/blob/db296ff7f47723980cb520f2bf98340dde096803/packages/web3-plugin-example/src/index.ts#L22)?

```ts
import { ContractAbi } from 'web3-eth-abi';
import Contract from 'web3-eth-contract';
import { Web3PluginBase } from 'web3-core';
import { Address, Web3APISpec } from 'web3-types';
// @ts-expect-error 'Web3' is declared but its value is never read.
import { Web3 } from 'web3/dist/web3';
```

Reply:
That doesn't really matter. The only difference here is if you put peer dependencies and set a broad range like ">=4.0.0 < 5" it will warn the user if he is supplying an invalid version, otherwise it would use his version. On the other hand, if you leave it as regular dep, it will install the plugin version of web3 along with a user-supplied version of web3.

Reply:
[This](https://github.com/web3/web3.js/pull/5393/commits/7642f41e41f78f9470fecb32ad43f70ab187239a) commit adds the dependencies as both `peerDependencies` and `devDependencies` as per [this](https://github.com/lerna/lerna/issues/375#issuecomment-326397282)